tools/omnissm/cmd/omnissm-agent/register.go

```diff
@@ -69,5 +55,7 @@ var RegisterCmd = &cobra.Command{
 
 func init() {
 	RegisterCmd.Flags().String("register-endpoint", "", "")
-	viper.BindPFlags(RegisterCmd.Flags())
+	RegisterCmd.LocalFlags().VisitAll(func(f *pflag.Flag) {
+		viper.BindPFlag(strings.Replace(f.Name, "-", "_", -1), RegisterCmd.Flags().Lookup(f.Name))
```

Comment:
I'm not entirely clear on intent, but it  looks like its updating to store cli -params into _params vars, but it seems a little odd that doesn't have a corresponding change to the variable usage.

Reply:
This was to get Viper to correctly associate the command line flag hyphenated `--register-endpoint` with the snakecase environment variable `OMNISSM_REGISTER_ENDPOINT`. so that both can be accessed with `viper.GetString("register_endpoint")`. Since we are no longer using a configuration file to select the appropriate endpoint, it is probably cleaner to remove Viper (at some point).